[CSS URL Request Modifiers] Support crossorigin() and referrerpolicy() modifiers on CSS image loads

weinig opened the pull request:
#### ea3e00e771db7d80e9fb207b881da3189913780a
<pre>
[CSS URL Request Modifiers] Support crossorigin() and referrerpolicy() modifiers on CSS image loads
<a href="https://bugs.webkit.org/show_bug.cgi?id=291379">https://bugs.webkit.org/show_bug.cgi?id=291379</a>

Reviewed by Darin Adler.

Adds support for applying crossorigin() and referrerpolicy() URL
request modifiers to CSS image loads.

The integrity() modifier is not fully implemented in this change
as it is not immediately clear where the best place to add the
integrity check in the current image loading pipeline would be.
It will added in a follow up. This does mean however that we
temporarily will start failing some of the parsing/serialization
tests that involve integrity() as we were falsely passing them
previously.

As only CSS images currently support the feature, parsing of the
modifiers has been disabled for all other consumers of the &lt;url&gt;
production via a new opt-in &quot;allowed modifiers&quot; parameter.

* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::loadImage):
    - Use the shared CSS::applyModifiersToLoaderOptions to apply the
      modifiers. This includes loadedFromOpaqueSource, which can now be
      removed from this call site.

* Source/WebCore/css/values/primitives/CSSURLModifiers.cpp:
(WebCore::CSS::applyModifiersToLoaderOptions):
    - Ensure fetch option credentials is set to &quot;same-origin&quot; for cors mode crossorigin(anonymous)
      image requests. This matches the spec (which currently says even no-cors requests should
      use &quot;same-origin&quot;, but that doesn&apos;t seem correct so we are not changing that) as well as
      our behavior for anonymous mask image requests in StylePendingResources.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Filter.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Masking.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Motion.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+URL.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+URL.h:
* Source/WebCore/css/scripts/process-css-properties.py:
    - Add new &quot;AllowedURLModifiers&quot; parameter for URL parsing and adopt it.

* Source/WebCore/css/scripts/test/TestCSSProperties.json:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyNames.gperf:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyNames.h:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyParsing.cpp:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSStyleProperties+PropertyNames.idl:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleBuilderGenerated.cpp:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleInterpolationWrapperMap.cpp:
    - Add tests for using the &quot;allowed-modifiers&quot; attribute on &lt;url&gt; productions.

* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-request-modifiers-computed.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-request-modifiers-serialize.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous-negative.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous-negative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials-negative.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials-negative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-green.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-navy.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-red.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/image-referrerpolicy.py: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/image-referrerpolicy.sub.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-image-ref.html: Added.
    - Add new tests.

Canonical link: <a href="https://commits.webkit.org/293647@main">https://commits.webkit.org/293647@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2f1f36f59cb8344f5db44686e16e142b54660bb7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99462 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19112 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9365 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104593 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50064 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101503 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19400 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27545 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75701 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32804 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102469 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14767 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89809 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56060 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14563 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7792 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49423 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84487 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7880 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106951 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26576 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19390 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84661 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26938 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86013 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84179 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21367 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28855 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6549 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20344 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26516 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31717 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26336 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29649 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27903 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+URL.h:
```diff
@@ -25,6 +25,7 @@
 #pragma once
 
 #include <wtf/Forward.h>
+#include <wtf/OptionSet.h>
```
I'm a little surprised that a forward declaration of `OptionSet` is not sufficient and already present in `Forward.h`.

weinig committed:
[CSS URL Request Modifiers] Support crossorigin() and referrerpolicy() modifiers on CSS image loads
https://bugs.webkit.org/show_bug.cgi?id=291379

Reviewed by Darin Adler.

Adds support for applying crossorigin() and referrerpolicy() URL
request modifiers to CSS image loads.

The integrity() modifier is not fully implemented in this change
as it is not immediately clear where the best place to add the
integrity check in the current image loading pipeline would be.
It will added in a follow up. This does mean however that we
temporarily will start failing some of the parsing/serialization
tests that involve integrity() as we were falsely passing them
previously.

As only CSS images currently support the feature, parsing of the
modifiers has been disabled for all other consumers of the <url>
production via a new opt-in "allowed modifiers" parameter.

* Source/WebCore/css/CSSImageValue.cpp:
(WebCore::CSSImageValue::loadImage):
    - Use the shared CSS::applyModifiersToLoaderOptions to apply the
      modifiers. This includes loadedFromOpaqueSource, which can now be
      removed from this call site.

* Source/WebCore/css/values/primitives/CSSURLModifiers.cpp:
(WebCore::CSS::applyModifiersToLoaderOptions):
    - Ensure fetch option credentials is set to "same-origin" for cors mode crossorigin(anonymous)
      image requests. This matches the spec (which currently says even no-cors requests should
      use "same-origin", but that doesn't seem correct so we are not changing that) as well as
      our behavior for anonymous mask image requests in StylePendingResources.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Filter.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Masking.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Motion.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+URL.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+URL.h:
* Source/WebCore/css/scripts/process-css-properties.py:
    - Add new "AllowedURLModifiers" parameter for URL parsing and adopt it.

* Source/WebCore/css/scripts/test/TestCSSProperties.json:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyNames.gperf:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyNames.h:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSPropertyParsing.cpp:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/CSSStyleProperties+PropertyNames.idl:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleBuilderGenerated.cpp:
* Source/WebCore/css/scripts/test/TestCSSPropertiesResults/StyleInterpolationWrapperMap.cpp:
    - Add tests for using the "allowed-modifiers" attribute on <url> productions.

* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-request-modifiers-computed.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-request-modifiers-serialize.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous-negative.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous-negative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-anonymous.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials-negative.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials-negative.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/cross-origin/url-image-crossorigin-use-credentials.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer-when-downgrade/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/no-referrer/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/same-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin-when-cross-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/strict-origin/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-cross-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-cross-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-same-origin.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/referrer-policy/unsafe-url/url-image-referrerpolicy-same-origin.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-green.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-navy.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/1x1-red.png: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/image-referrerpolicy.py: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/support/image-referrerpolicy.sub.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/urls/url-image-ref.html: Added.
    - Add new tests.

Canonical link: https://commits.webkit.org/293647@main
